components/prospects/ProspectNotes.tsx

```diff
@@ -0,0 +1,225 @@
+      method: 'PUT',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ id: noteId, note: newNote })
+    })
+    .then(() => {
+      const updatedNoteIndex = notes.findIndex((note) => note.id === noteId)
+      const updatedNote = { ...notes[updatedNoteIndex], note: newNote }
+      const updatedNotes = [...notes]
+      updatedNotes[updatedNoteIndex] = updatedNote
+      setNotes(updatedNotes)
+      setSelectedNote(null)
+    })
+    .catch((error) => console.error('Error:', error))
+  }
+
+  function handleNoteChange(event) {
+    const newNote = event.target.value
+    setSelectedNote((prevNote) => ({ ...prevNote, note: newNote }))
+  }
+
+  function handleSaveNote() {
+    if (selectedNote) {
+      updateNote(selectedNote.id, selectedNote.note)
+    }
+    setSelectedNote(null)
+    setIsEditing(false)
+  }
+
+  function handleCancelNote() {
+    setSelectedNote(null)
+    setIsEditing(false)
+  }
+
+  function handleEditNote(note) {
+    setSelectedNote(note)
+    setIsEditing(true)
+  }
+
+  function handleDeleteNote(note) {
+    setNoteToDelete(note)
+    setDeleteModalOpen(true)
+  }
+
+  function handleConfirmDelete() {
+    if (noteToDelete) {
+      deleteNote(noteToDelete.id)
+    }
+    setDeleteModalOpen(false)
+    setNoteToDelete(null)
+  }
+
+  function handleCancelDelete() {
+    setDeleteModalOpen(false)
+    setNoteToDelete(null)
+  }
+
+  function formatDateTime(dateTime) {
+    const date = new Date(dateTime)
+    const day = date.getDate()
+    const month = date.getMonth() + 1
+    const year = date.getFullYear()
+    const hours = date.getHours()
+    const minutes = date.getMinutes()
+    const seconds = date.getSeconds()
+
+    const formattedDay = day < 10 ? `0${day}` : day
+    const formattedMonth = month < 10 ? `0${month}` : month
+    const formattedHours = hours < 10 ? `0${hours}` : hours
+    const formattedMinutes = minutes < 10 ? `0${minutes}` : minutes
+    const formattedSeconds = seconds < 10 ? `0${seconds}` : seconds
+
+    return `${formattedDay}/${formattedMonth}/${year} 
+      ${formattedHours}:${formattedMinutes}:${formattedSeconds}`
+  }
+
+  return (
+    <Box>
+      {!selectedNote && (
+        <NoteTextarea prospectId={prospectId} />
+      )}
+      {notes.map((note) => (
+        <Card key={note.id}>
+          <CardContent>
+            {selectedNote && selectedNote.id === note.id ? (
+              <Modal
+                open={isEditing}
+                onClose={handleCancelNote}>
+                <Box
+                  sx={{
+                    position: 'absolute',
+                    top: '50%',
+                    left: '50%',
+                    transform: 'translate(-50%, -50%)',
+                    bgcolor: 'background.paper',
+                    border: '1px solid #000',
+                    boxShadow: 24,
+                    p: '2rem',
+                    width: '80vw'
+                  }}>
+                  <TextField
```

Comment:
we should have a single component for updating / creating a note in order to avoid repetitive code